crates/re_viewer/src/ui/rerun_menu.rs

```diff
@@ -154,42 +162,31 @@ fn recordings_menu(ui: &mut egui::Ui, store_hub: &StoreHub, app_state: &mut AppS
             "<UNKNOWN>".to_owned()
         };
         if ui
-            .radio(
-                app_state.recording_id().as_ref() == Some(store_db.store_id()),
-                info,
-            )
+            .radio(active_recording == Some(store_db.store_id()), info)
             .clicked()
         {
-            app_state.set_recording_id(store_db.store_id().clone());
+            // TODO(jleibs): This is gross but necessary to avoid a deadlock
+            app.requested_recording_id = Some(store_db.store_id().clone());
```

Comment:
Could that request be stored in the store hub instead of the app? This way the recording selection UI would only be reliant on store hub, which would make it easier to move it deeper in the UI code.

Reply:
Resolved in: https://github.com/rerun-io/rerun/pull/2344